include/jwt-cpp/jwt.h

```diff
@@ -2463,7 +2463,8 @@ namespace jwt {
 			static constexpr auto value = std::is_constructible<value_type, array_type>::value &&
 										  is_iterable<array_type>::value &&
 										  is_detected<value_type_t, array_type>::value &&
-										  std::is_same<typename array_type::value_type, value_type>::value;
+										  std::is_same<typename array_type::value_type, value_type>::value &&
+										  !std::is_same<decltype(std::declval<array_type>().front()), void>::value;
```

Comment:
Hmm this is a good method but I am curious is the return should be `value_type_t` rather them not void?

I generally see these written separately so the name is a bit more meaningful

https://github.com/Thalhammer/jwt-cpp/blob/feaca3cc8541e49b3508fdd07abe83851f858c91/include/jwt-cpp/jwt.h#L2430

Reply:
But `value_type_t` is templated, wouldn't this make `value` templated as well?
Really not sure how `is_detected` using it or what does it do.

Anyway, I changed it to `std::is_member_function_pointer` since I removed the non-const version.